### ISSUE TITLE
MongoDB duplicate all fields into an Collection

### ISSUE THREAD

**Issue by edumaciel10** [metabase/metabase#30876]
### Describe the bug

My Metabase instance break sometimes into this week, and into this last week before break, the Company collection appears with a lot of duplicated data, but I don't know how to debug it to explain how to reproduce
![Screen Shot 2023-05-19 at 10 14 59](https://github.com/metabase/metabase/assets/54045014/a2c4bb92-e22d-45d2-86ca-81a44aa20640)

### Expected behavior

Not duplicate _id into this company

### Logs

```
[38881769-3425-4439-946b-17bf52ffcbd5] 2023-05-18T22:59:19-03:00 WARN metabase.sync.util Error fingerprinting Table 447 'Company'
com.mongodb.MongoCommandException: Command failed with error 16007 (Location16007): 'PlanExecutor error during aggregation :: caused by :: can't convert from BSON type array to String' on server mongodb-all-nodes.****-mongo-stg:27017. The full response is {"ok": 0.0, "errmsg": "PlanExecutor error during aggregation :: caused by :: can't convert from BSON type array to String", "code": 16007, "codeName": "Location16007", "$clusterTime": {"clusterTime": {"$timestamp": {"t": 1684461550, "i": 1}}, "signature": {"hash": {"$binary": "AAAAAAAAAAAAAAAAAAAAAAAAAAA=", "$type": "00"}, "keyId": {"$numberLong": "0"}}}, "operationTime": {"$timestamp": {"t": 1684461550, "i": 1}}}
	at com.mongodb.internal.connection.ProtocolHelper.getCommandFailureException(ProtocolHelper.java:175)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveCommandMessageResponse(InternalStreamConnection.java:302)
	at com.mongodb.internal.connection.InternalStreamConnection.sendAndReceive(InternalStreamConnection.java:258)
	at com.mongodb.internal.connection.UsageTrackingInternalConnection.sendAndReceive(UsageTrackingInternalConnection.java:99)
	at com.mongodb.internal.connection.DefaultConnectionPool$PooledConnection.sendAndReceive(DefaultConnectionPool.java:450)
	at com.mongodb.internal.connection.CommandProtocolImpl.execute(CommandProtocolImpl.java:72)
	at com.mongodb.internal.connection.DefaultServer$DefaultServerProtocolExecutor.execute(DefaultServer.java:226)
	at com.mongodb.internal.connection.DefaultServerConnection.executeProtocol(DefaultServerConnection.java:269)
	at com.mongodb.internal.connection.DefaultServerConnection.command(DefaultServerConnection.java:131)
	at com.mongodb.internal.connection.DefaultServerConnection.command(DefaultServerConnection.java:123)
	at com.mongodb.operation.CommandOperationHelper.executeCommand(CommandOperationHelper.java:343)
	at com.mongodb.operation.CommandOperationHelper.executeCommand(CommandOperationHelper.java:334)
	at com.mongodb.operation.CommandOperationHelper.executeCommandWithConnection(CommandOperationHelper.java:220)
	at com.mongodb.operation.CommandOperationHelper$5.call(CommandOperationHelper.java:206)
	at com.mongodb.operation.OperationHelper.withReadConnectionSource(OperationHelper.java:463)
	at com.mongodb.operation.CommandOperationHelper.executeCommand(CommandOperationHelper.java:203)
	at com.mongodb.operation.AggregateOperationImpl.execute(AggregateOperationImpl.java:200)
	at com.mongodb.operation.AggregateOperation.execute(AggregateOperation.java:333)
	at com.mongodb.operation.AggregateOperation.execute(AggregateOperation.java:42)
	at com.mongodb.client.internal.MongoClientDelegate$DelegateOperationExecutor.execute(MongoClientDelegate.java:199)
	at com.mongodb.client.internal.MongoClientDelegate$DelegateOperationExecutor.execute(MongoClientDelegate.java:180)
	at com.mongodb.DBCollection.aggregate(DBCollection.java:1530)
	at com.mongodb.DBCollection.aggregate(DBCollection.java:1488)
	at com.mongodb.DBCollection.aggregate(DBCollection.java:1473)
	at metabase.driver.mongo.execute$aggregate.invokeStatic(execute.clj:142)
	at metabase.driver.mongo.execute$aggregate.invoke(execute.clj:136)
	at metabase.driver.mongo.execute$execute_reducible_query.invokeStatic(execute.clj:179)
	at metabase.driver.mongo.execute$execute_reducible_query.invoke(execute.clj:173)
	at metabase.driver.mongo$fn__120607$f__119429__auto____120608.invoke(mongo.clj:308)
	at metabase.driver.mongo$fn__120607.invokeStatic(mongo.clj:307)
	at metabase.driver.mongo$fn__120607.invoke(mongo.clj:305)
	at clojure.lang.MultiFn.invoke(MultiFn.java:244)
	at metabase.query_processor.context$executef.invokeStatic(context.clj:60)
	at metabase.query_processor.context$executef.invoke(context.clj:49)
	at metabase.query_processor.context.default$default_runf.invokeStatic(default.clj:68)
	at metabase.query_processor.context.default$default_runf.invoke(default.clj:66)
	at metabase.query_processor.context$runf.invokeStatic(context.clj:46)
	at metabase.query_processor.context$runf.invoke(context.clj:40)
	at metabase.query_processor.reducible$identity_qp.invokeStatic(reducible.clj:15)
	at metabase.query_processor.reducible$identity_qp.invoke(reducible.clj:12)
	at metabase.query_processor.middleware.cache$maybe_return_cached_results$maybe_return_cached_results_STAR___69172.invoke(cache.clj:224)
	at metabase.query_processor.middleware.permissions$check_query_permissions$fn__64644.invoke(permissions.clj:126)
	at metabase.query_processor.middleware.mbql_to_native$mbql__GT_native$fn__68063.invoke(mbql_to_native.clj:24)
	at metabase.query_processor$fn__70691$combined_post_process__70696$combined_post_process_STAR___70697.invoke(query_processor.clj:243)
	at metabase.query_processor$fn__70691$combined_pre_process__70692$combined_pre_process_STAR___70693.invoke(query_processor.clj:240)
	at metabase.query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__69083$fn__69088.invoke(resolve_database_and_driver.clj:36)
	at metabase.driver$do_with_driver.invokeStatic(driver.clj:90)
	at metabase.driver$do_with_driver.invoke(driver.clj:86)
	at metabase.query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__69083.invoke(resolve_database_and_driver.clj:35)
	at metabase.query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__64953.invoke(fetch_source_query.clj:310)
	at metabase.query_processor.middleware.store$initialize_store$fn__65131$fn__65132.invoke(store.clj:12)
	at metabase.query_processor.store$do_with_store.invokeStatic(store.clj:45)
	at metabase.query_processor.store$do_with_store.invoke(store.clj:41)
	at metabase.query_processor.middleware.store$initialize_store$fn__65131.invoke(store.clj:11)
	at metabase.query_processor.middleware.normalize_query$normalize$fn__69372.invoke(normalize_query.clj:25)
	at metabase.query_processor.reducible$async_qp$qp_STAR___59455$thunk__59457.invoke(reducible.clj:103)
	at metabase.query_processor.reducible$async_qp$qp_STAR___59455.invoke(reducible.clj:109)
	at metabase.query_processor.reducible$async_qp$qp_STAR___59455.invoke(reducible.clj:94)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.query_processor.reducible$sync_qp$qp_STAR___59467.doInvoke(reducible.clj:129)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:669)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.query_processor$process_query.invokeStatic(query_processor.clj:278)
	at metabase.query_processor$process_query.doInvoke(query_processor.clj:272)
	at clojure.lang.RestFn.invoke(RestFn.java:423)
	at clojure.lang.Var.invoke(Var.java:388)
	at metabase.db.metadata_queries$fn__77208$table_rows_sample__77217$fn__77220.invoke(metadata_queries.clj:145)
	at metabase.db.metadata_queries$fn__77208$table_rows_sample__77217.invoke(metadata_queries.clj:132)
	at metabase.driver.mongo$fn__120635.invokeStatic(mongo.clj:365)
	at metabase.driver.mongo$fn__120635.invoke(mongo.clj:361)
	at clojure.lang.MultiFn.invoke(MultiFn.java:252)
	at metabase.sync.analyze.fingerprint$fn__85045$fingerprint_table_BANG___85050$fn__85051.invoke(fingerprint.clj:76)
	at metabase.sync.analyze.fingerprint$fn__85045$fingerprint_table_BANG___85050.invoke(fingerprint.clj:54)
	at metabase.sync.analyze.fingerprint$fn__85226$fingerprint_fields_BANG___85231$fn__85232$fn__85233.invoke(fingerprint.clj:190)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:161)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:154)
	at metabase.sync.analyze.fingerprint$fn__85226$fingerprint_fields_BANG___85231$fn__85232.invoke(fingerprint.clj:188)
	at metabase.sync.analyze.fingerprint$fn__85226$fingerprint_fields_BANG___85231.invoke(fingerprint.clj:184)
	at metabase.sync.analyze.fingerprint$fn__85262$fingerprint_fields_for_db_BANG__STAR___85271$fn__85274$fn__85275$fn__85276.invoke(fingerprint.clj:214)
	at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
	at clojure.core$reduce.invokeStatic(core.clj:6885)
	at clojure.core$reduce.invoke(core.clj:6868)
	at metabase.sync.analyze.fingerprint$fn__85262$fingerprint_fields_for_db_BANG__STAR___85271$fn__85274$fn__85275.invoke(fingerprint.clj:210)
	at metabase.query_processor.store$do_with_store.invokeStatic(store.clj:47)
	at metabase.query_processor.store$do_with_store.invoke(store.clj:41)
	at metabase.sync.analyze.fingerprint$fn__85262$fingerprint_fields_for_db_BANG__STAR___85271$fn__85274.invoke(fingerprint.clj:207)
	at metabase.sync.analyze.fingerprint$fn__85262$fingerprint_fields_for_db_BANG__STAR___85271.invoke(fingerprint.clj:196)
	at metabase.sync.analyze.fingerprint$fn__85262$fingerprint_fields_for_db_BANG__STAR___85271$fn__85272.invoke(fingerprint.clj:201)
	at metabase.sync.analyze.fingerprint$fn__85262$fingerprint_fields_for_db_BANG__STAR___85271.invoke(fingerprint.clj:196)
	at metabase.sync.analyze.fingerprint$fn__85305$fingerprint_fields_for_db_BANG___85310$fn__85311.invoke(fingerprint.clj:228)
	at metabase.sync.analyze.fingerprint$fn__85305$fingerprint_fields_for_db_BANG___85310.invoke(fingerprint.clj:222)
	at metabase.sync.analyze$make_analyze_steps$fn__85658.invoke(analyze.clj:105)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:669)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.sync.util$fn__55915$run_step_with_metadata__55920$fn__55924$fn__55926.doInvoke(util.clj:379)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:103)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:97)
	at metabase.sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:120)
	at metabase.sync.util$with_start_and_finish_debug_logging.invoke(util.clj:117)
	at metabase.sync.util$fn__55915$run_step_with_metadata__55920$fn__55924.invoke(util.clj:374)
	at metabase.sync.util$fn__55915$run_step_with_metadata__55920.invoke(util.clj:369)
	at metabase.sync.util$fn__56142$run_sync_operation__56147$fn__56148$fn__56156.invoke(util.clj:486)
	at metabase.sync.util$fn__56142$run_sync_operation__56147$fn__56148.invoke(util.clj:484)
	at metabase.sync.util$fn__56142$run_sync_operation__56147.invoke(util.clj:478)
	at metabase.sync.analyze$fn__85671$analyze_db_BANG___85676$fn__85677$fn__85678.invoke(analyze.clj:122)
	at metabase.sync.util$do_with_error_handling.invokeStatic(util.clj:161)
	at metabase.sync.util$do_with_error_handling.invoke(util.clj:154)
	at clojure.core$partial$fn__5910.invoke(core.clj:2647)
	at metabase.driver.mongo$fn__120463$f__119429__auto____120464.invoke(mongo.clj:104)
	at metabase.driver.mongo.util$do_with_mongo_connection$fn__119417.invoke(util.clj:230)
	at metabase.util.ssh$do_with_ssh_tunnel.invokeStatic(ssh.clj:141)
	at metabase.util.ssh$do_with_ssh_tunnel.invoke(ssh.clj:130)
	at metabase.driver.mongo.util$do_with_mongo_connection.invokeStatic(util.clj:224)
	at metabase.driver.mongo.util$do_with_mongo_connection.invoke(util.clj:219)
	at metabase.driver.mongo$fn__120463.invokeStatic(mongo.clj:103)
	at metabase.driver.mongo$fn__120463.invoke(mongo.clj:101)
	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
	at metabase.sync.util$sync_in_context$fn__55835.invoke(util.clj:139)
	at metabase.sync.util$with_db_logging_disabled$fn__55832.invoke(util.clj:130)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invokeStatic(util.clj:103)
	at metabase.sync.util$with_start_and_finish_logging_STAR_.invoke(util.clj:97)
	at metabase.sync.util$with_start_and_finish_logging$fn__55819.invoke(util.clj:115)
	at metabase.sync.util$with_sync_events$fn__55814.invoke(util.clj:89)
	at metabase.sync.util$with_duplicate_ops_prevented$fn__55804.invoke(util.clj:68)
	at metabase.sync.util$do_sync_operation.invokeStatic(util.clj:182)
	at metabase.sync.util$do_sync_operation.invoke(util.clj:179)
	at metabase.sync.analyze$fn__85671$analyze_db_BANG___85676$fn__85677.invoke(analyze.clj:119)
	at metabase.sync.analyze$fn__85671$analyze_db_BANG___85676.invoke(analyze.clj:114)
	at metabase.task.sync_databases$sync_and_analyze_database_BANG_.invokeStatic(sync_databases.clj:79)
	at metabase.task.sync_databases$sync_and_analyze_database_BANG_.invoke(sync_databases.clj:67)
	at metabase.task.sync_databases.SyncAndAnalyzeDatabase.execute(sync_databases.clj:86)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
[38881769-3425-4439-946b-17bf52ffcbd5] 2023-05-18T22:59:19-03:00 INFO metabase.sync.analyze fingerprint-fields Analyzed [***************···································] 😕   30% Table 471 'CompanyCandidateDocument'
```

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.19+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.19",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.19+7",
    "os.name": "Linux",
    "os.version": "5.4.226-129.415.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "mongo"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "2.1.212 (2022-04-09)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "2.1.212 (2022-04-09)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-04-28",
      "tag": "v0.46.2",
      "branch": "release-x.46.x",
      "hash": "8967c94"
    },
    "settings": {
      "report-timezone": "America/Sao_Paulo"
    }
  }
}
```

### Severity

high, this confuse my team to create questions, because the name of fields are duplicated, and we can't assert the right field

**Comment by paoliniluis** [comment]
It would be awesome to know how you reached that state, seems that there's a collection where there's a projection (maybe a view?) 
"can't convert from BSON type array to String'"

**Comment by paoliniluis** [comment]
Closing due to non response

**Comment by moroine** [comment]
@paoliniluis I've ran into the same issue with duplicated field. I don't know how I ended-up there (I'm new on the project) but looks like we have duplicated field when using nested field.

My database was using H2, but I tried to migrate to Postgres and ran into this issue https://github.com/metabase/metabase/issues/10087. I disabled the constraint and it worked. Then I looked at the metadata fields and I have duplicated `table_id + name` but with different `parent_id`.

My guess, is this comes from this kind of document would cause the issue:
```js
{
   _id: ObjectId('xxx'),
   data: { _id: ObjectId('xxx') },
}
```
